### ISSUE TITLE
composer update --no-dev makes changes on newly installed project 

### ISSUE THREAD

**Issue by melato** [drupal-composer/drupal-composer-project#444]
I did this:

```
composer create-project drupal-composer/drupal-project:8.x-dev --no-dev --stability dev --no-interaction d1
cd d1
composer update --no-dev
```

The update did this:

```
Package operations: 1 install, 2 updates, 0 removals
  - Installing symfony/polyfill-php72 (v1.10.0): Downloading (100%)         
  - Updating symfony/var-dumper (v3.4.20 => v4.2.1): Downloading (100%)         
  - Downgrading drush/drush (9.5.2 => 9.4.0): Downloading (100%)         
```
What is going on here?
Why didn't create-project install the final versions from the start?
I tried this a few times, with clean cache, etc.

If this is a composer bug, let me know if I should enter it there.

**Comment by webflo** [comment]
Its not a composer problem. This happens because the drush has less strict dependencies in version `9.4.0` than in `9.5.2`. If you want to build a site without the dev packages you should run `composer install --no-dev`. Or use a stricter version constraint for drush.

This issue will be resolved once drush 9.6.0 has been released, because it supports Symfony 3 and 4.

**Comment by gitressa** [comment]
I just got hit by this, and am looking forward to drush `9.6.0`. Setting version constraint on drush version, as a temporary solution, fixed it:

`composer require --update-no-dev 'drush/drush:~9.5.0' -v`

I had to use tilde (`~`) not caret (`^`), to stay on drush 9.5.x:

**"drush/drush": "^9.5.0" in composer.json**
Will update drush to 9.6.0-rc3
```
$ composer update --no-dev --with-dependencies --dry-run -v
Loading composer repositories with package information
Updating dependencies                                    
Dependency resolution completed in 3.008 seconds
Analyzed 16358 packages to resolve dependencies
Analyzed 1544441 rules to resolve dependencies
Dependency resolution completed in 0.000 seconds
Package operations: 3 installs, 3 updates, 0 removals
Installs: symfony/polyfill-php72:v1.10.0, consolidation/site-process:1.1.2, consolidation/filter-via-dot-access-data:1.0.0
Updates: symfony/var-dumper:v4.2.3, consolidation/site-alias:2.1.1, drush/drush:9.6.0-rc3
  - Installing symfony/polyfill-php72 (v1.10.0)
  - Updating symfony/var-dumper (v3.4.22) to symfony/var-dumper (v4.2.3)
  - Updating consolidation/site-alias (1.1.11) to consolidation/site-alias (2.1.1)
  - Installing consolidation/site-process (1.1.2)
  - Installing consolidation/filter-via-dot-access-data (1.0.0)
  - Updating drush/drush (9.5.2) to drush/drush (9.6.0-rc3)
```

**"drush/drush": "~9.5.0"**
Keeps drush 9.5.2
```
$ composer update --no-dev --with-dependencies --dry-run -v
Loading composer repositories with package information
Updating dependencies                                    
Dependency resolution completed in 17.627 seconds
Analyzed 16292 packages to resolve dependencies
Analyzed 1543885 rules to resolve dependencies
Dependency resolution completed in 0.000 seconds
```
I could add a hard constraint like `"drush/drush": "9.5.2"` but went for `"drush/drush": "~9.5.0"` just in case there is a `9.5.3` released ...
